python/lsst/ts/ledprojector/led_controller.py

```diff
@@ -0,0 +1,319 @@
+# This file is part of ts_ledprojector.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["LEDController"]
+import asyncio
+import inspect
+import logging
+import types
+
+from lsst.ts import salobj, utils
+from lsst.ts.idl.enums.run_ledprojector import (
+    LEDBasicState,
+)
+from .labjack_interface import LabJackInterface
+# Hide mypy error `Module "labjack" has no attribute "ljm"`.
+from labjack import ljm  # type: ignore
+from .mock_labjack_interface import MockLabJackInterface
+from .base_labjack_data_client import BaseLabJackDataClient
+
+# DRA-CN024D05 SSR requires 3.0VDC minimum turn-on voltage, up to 12 VDC max
+# 1.0VDC minimum turn-off voltage
+# all ports should be digital IO so write 1 for on/0 for off
+# Values to write to labjack for operating LED
+class LEDSetting():
+    def __init__():
+        LED_OFF_SETTING = 0
+        LED_ON_SETTING = 1
+
+# Time limit for communicating with the LabJack (seconds).
+COMMUNICATION_TIMEOUT = 5
+
+# Time limit for configuring streaming;
+# measured time from far away is 6 seconds.
+START_STREAMING_TIMEOUT = 15
+
+# Maximum frequency (Hz) at which the mock simulator can read an input.
+# The max read frequency per channel = MAX_MOCK_READ_FREQUENCY / num channels.
+MAX_MOCK_READ_FREQUENCY = 10000
+
+# Number of channels per accelerometer
+# We assume a 3-axis accelerometr: x, y, z
+NUM_CHANNELS_PER_ACCELEROMETER = 3
+
+# Number of accelerometer arrays full of random data to auto-generate.
+# 5 gives a few unique accelerometer messages to cycle through,
+# without using a crazy amount of memory.
+NUM_RANDOM_ACCEL_ARRAYS = 5
+
+class LabJackChannels():
```

Comment:
Consider using a `enum.StrEnum` instead of having a class with a dictionary. It would look something like this:

```

class LabJackChannels(enum.StrEnum):
    EI00 = "M375L4"
    EI01 = "M445L4"
    EI02 = "M505L4"
    EI03 = "M565L4"
    EI05 = "M660L4"
    EI06 = "M730L5"
    EI07 = "M780LP1"
    EI08 = "M850L3"
    EI09 = "M940L3"
    EI10 = "M970L4"
```

Also, consider moving this to separate submodule and make the comment lines below be part of the class docstring.